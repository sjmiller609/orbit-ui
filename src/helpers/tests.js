import { BrowserRouter } from 'react-router-dom'
import { shape } from 'prop-types'
import { mount } from 'enzyme'

// Instantiate router context
const router = {
  history: new BrowserRouter().history,
  route: {
    location: {},
    match: {},
  },
}

const createContext = () => ({
  context: { router },
  childContextTypes: { router: shape({}) },
})

// wrap with router context - to mount RR links correctly
export function mountWrap(node) {
  return mount(node, createContext())
}
