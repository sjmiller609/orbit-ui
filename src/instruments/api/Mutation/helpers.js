const has = (str, find) => ~str.toLowerCase().indexOf(find.toLowerCase())

export const formErrors = ({ error, errors }) => {
  if (!error) return

  let err = JSON.stringify(error).toLowerCase()
  return errors.find(e => {
    if (e.key && has(err, e.key)) return e
    else if (e.allKeys && e.allKeys.every(k => has(err, k))) return e
    else if (e.someKeys && e.someKeys.some(k => has(err, k))) return e
  })
}
