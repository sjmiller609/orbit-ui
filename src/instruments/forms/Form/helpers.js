// faltten and stringify nested keys
export const unpack = (obj, ns, hash) => {
  if (!obj) return
  return Object.entries(obj).reduce((acc, [key, value]) => {
    const keypath = ns ? `${ns}.${key}` : key
    if (value && typeof value === 'object') {
      return unpack(value, keypath, acc)
    }
    acc[keypath] = value
    return acc
  }, hash || {})
}

const roll = (keys, obj, value) => {
  const key = keys[0]
  // case 1 - not obj
  if (keys.length === 1) {
    obj[key] = value
    return obj
  }
  // case 2 - doesn't exist
  if (!obj[key]) {
    obj[key] = isNaN(keys[1]) ? {} : []
  }
  // case 3 - exists
  obj[key] = roll(
    keys.slice(1),
    Array.isArray(obj[key]) ? [...obj[key]] : { ...obj[key] },
    value
  )

  return obj
}

// convert back into obj
export const pack = (obj, type = {}) => {
  return Object.entries(obj).reduce((acc, [key, value]) => {
    const keys = key.split('.')

    return roll(keys, acc, value)
  }, type)
}

export const packChild = ({ name, obj }) => {
  const obj2 = {}
  let isArray
  Object.entries(obj).forEach(k => {
    if (name.length < k[0].length && k[0].indexOf(name) === 0) {
      const n = k[0].slice(name.length + 1)
      // must be true for all matching
      isArray = !isNaN(n.slice(0, n.indexOf('.')))
      obj2[n] = k[1]
    }
  })
  if (Object.keys(obj2).length) return pack(obj2, isArray ? [] : {})
  return null
}

export const removeChild = ({ name, obj = {} }) => {
  const obj2 = {}
  Object.entries(obj).forEach(k => {
    if (name.length >= k[0].length || k[0].indexOf(name) !== 0) {
      obj2[k[0]] = k[1]
    }
  })
  return obj2
}
