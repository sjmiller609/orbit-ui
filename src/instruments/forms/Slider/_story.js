import React from 'react'
import { storiesOf } from '@storybook/react'
import { Slider } from 'instruments'

storiesOf('Instruments|Forms.Slider', module).add('Default', () => (
  <Slider min={1} max={10} value={5} step={1} onChange={() => null} />
))
