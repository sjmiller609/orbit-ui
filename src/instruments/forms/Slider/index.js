import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import { default as RCSlider } from 'rc-slider'

import s from './styles.scss'

class Slider extends React.Component {
  render() {
    const { className, disabled, ...props } = this.props
    return (
      <RCSlider
        {...props}
        className={classnames(
          s.slider,
          disabled ? s.disabled : null,
          className
        )}
      />
    )
  }
}

Slider.propTypes = {
  min: PropTypes.number,
  max: PropTypes.number,
  value: PropTypes.number,
  step: PropTypes.number,
  onChange: PropTypes.func,
  className: PropTypes.string,
  disabled: PropTypes.bool,
}

export default Slider
