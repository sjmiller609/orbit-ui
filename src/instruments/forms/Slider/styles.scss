@import 'src/styles/mixins.scss';

.slider {
  position: relative;
  height: 1.5rem;
  width: 100%;
  :global {
    .rc-slider-rail {
      position: absolute;
      width: 100%;
      background-color: $darkWhite;
      height: 100%;
      border-radius: 100px;
      //  box-shadow: inset 0 1px 3px $shadow;
    }
    .rc-slider-track {
      position: absolute;
      left: 0;
      height: 100%;
      border-radius: 100px;
      //  box-shadow: inset 0 1px 3px $shadow;
      background: linear-gradient(to right, $green, $lightGreen);
    }
    .rc-slider-handle {
      position: absolute;
      margin-left: -1.25rem;
      margin-top: -0.5rem;
      width: 2.5rem;
      height: 2.5rem;
      cursor: pointer;
      cursor: -webkit-grab;
      cursor: grab;
      background: $white;
      border-radius: 100rem;
      cursor: pointer;
      display: inline-block;
      box-shadow: 0.1rem 0.3rem 3.3rem 0.1rem $shadow, inset 0 0 0.5px 3px $gray;
      outline: none;
      touch-action: pan-x;
    }
    .disabled {
      background-image: none;
      opacity: 0.6;
      pointer-events: none;
    }
  }
}
