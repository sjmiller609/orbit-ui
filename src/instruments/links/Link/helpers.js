// check if url is external
export const externalUrl = url => {
  if (typeof url === 'object') return false
  const path = /^https?:\/\/|^\/\//i
  if (path.test(url) && !~url.indexOf(window.location.origin)) return true
  return
}
