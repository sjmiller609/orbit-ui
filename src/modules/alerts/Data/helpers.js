import { formErrors } from 'instruments'

const errors = [
  {
    key: 'permissions',
    name: 'properties.alert_emails',
    error: 'Contact your workspace admin to upgrade your permissions.',
  },
]

export const handleError = error => formErrors({ error, errors })

export const trimError = error => error.split(':')[1].trim()
