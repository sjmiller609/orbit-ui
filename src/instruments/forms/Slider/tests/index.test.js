import React from 'react'
import { mountWrap } from 'helpers/tests'
import Slider from '../index'

const props = {
  min: 1,
  max: 10,
  value: 5,
  step: 1,
  onChange: jest.fn(),
  className: 'testClassString',
  disabled: false,
}

describe('Slider', () => {
  it('renders correctly', () => {
    const wrapper = mountWrap(<Slider {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
})
