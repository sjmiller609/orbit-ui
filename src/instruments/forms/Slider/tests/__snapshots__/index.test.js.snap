// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider renders correctly 1`] = `
<Slider
  className="testClassString"
  disabled={false}
  max={10}
  min={1}
  onChange={[Function]}
  step={1}
  value={5}
>
  <ComponentEnhancer(undefined)
    activeDotStyle={Object {}}
    className="slider testClassString"
    disabled={false}
    dotStyle={Object {}}
    dots={false}
    handle={[Function]}
    handleStyle={
      Array [
        Object {},
      ]
    }
    included={true}
    marks={Object {}}
    max={10}
    min={1}
    onAfterChange={[Function]}
    onBeforeChange={[Function]}
    onChange={[Function]}
    prefixCls="rc-slider"
    railStyle={Object {}}
    step={1}
    trackStyle={
      Array [
        Object {},
      ]
    }
    value={5}
    vertical={false}
  >
    <div
      className="rc-slider slider testClassString"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="rc-slider-rail"
        style={Object {}}
      />
      <Track
        className="rc-slider-track"
        included={true}
        length={44.44444444444444}
        offset={0}
        style={Object {}}
        vertical={false}
      >
        <div
          className="rc-slider-track"
          style={
            Object {
              "left": "0%",
              "width": "44.44444444444444%",
            }
          }
        />
      </Track>
      <Steps
        activeDotStyle={Object {}}
        dotStyle={Object {}}
        dots={false}
        included={true}
        lowerBound={1}
        marks={Object {}}
        max={10}
        min={1}
        prefixCls="rc-slider"
        step={1}
        upperBound={5}
        vertical={false}
      >
        <div
          className="rc-slider-step"
        />
      </Steps>
      <Handle
        className="rc-slider-handle"
        disabled={false}
        key="0"
        max={10}
        min={1}
        offset={44.44444444444444}
        style={Object {}}
        value={5}
        vertical={false}
      >
        <div
          aria-disabled={false}
          aria-valuemax={10}
          aria-valuemin={1}
          aria-valuenow={5}
          className="rc-slider-handle"
          role="slider"
          style={
            Object {
              "left": "44.44444444444444%",
            }
          }
          tabIndex={0}
        />
      </Handle>
      <Marks
        className="rc-slider-mark"
        included={true}
        lowerBound={1}
        marks={Object {}}
        max={10}
        min={1}
        upperBound={5}
        vertical={false}
      >
        <div
          className="rc-slider-mark"
        />
      </Marks>
    </div>
  </ComponentEnhancer(undefined)>
</Slider>
`;
